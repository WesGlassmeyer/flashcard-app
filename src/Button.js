import React from "react";

export default function Button({ text, style, handleClick, handleNoClick }) {
  return (
    <button className={style} onClick={handleClick || handleNoClick}>
      {text}
    </button>
  );
}
