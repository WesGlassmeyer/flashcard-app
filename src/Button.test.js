import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Button from "./Button";
import toJSON from "enzyme-to-json";
import { isExportDeclaration } from "typescript";

describe("Button component", () => {
  const props = {
    text: "text",
    style: "color",
  };
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders a button by default", () => {
    const wrapper = shallow(<Button />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders a button with props", () => {
    const wrapper = shallow(<Button {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
