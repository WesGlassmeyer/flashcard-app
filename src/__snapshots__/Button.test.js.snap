// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders a button by default 1`] = `<button />`;

exports[`Button component renders a button with props 1`] = `
<button
  className="color"
>
  text
</button>
`;
